src/mcp_atlassian/server.py

```diff
@@ -17,6 +21,79 @@
 logger = logging.getLogger("mcp-atlassian")
 
 
+def configure_logging() -> None:
+    """Configure logging based on environment variables."""
+    debug_mode = os.getenv("MCP_ATLASSIAN_DEBUG", "0") == "1"
+
+    # Set up logging format
+    log_format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+    if debug_mode:
+        logging.basicConfig(level=logging.DEBUG, format=log_format)
+        logger.setLevel(logging.DEBUG)
+        logger.debug("Debug mode enabled")
+    else:
+        logging.basicConfig(level=logging.INFO, format=log_format)
+        logger.setLevel(logging.INFO)
+
+        # Set MCP core loggers to WARNING level to reduce noise
+        # This will prevent the standard request processing logs
+        logging.getLogger("mcp.server.lowlevel.server").setLevel(logging.WARNING)
+        logging.getLogger("mcp.server").setLevel(logging.WARNING)
+
+
+# Call configure_logging at import time
+configure_logging()
+
+
+class DebugTransport:
+    """Wrapper for transport that logs all messages."""
+
+    def __init__(self, read_stream: Any, write_stream: Any) -> None:
+        self.read_stream = read_stream
+        self.write_stream = write_stream
+
+    async def read(self) -> str:
+        msg = await self.read_stream.read()
+        logger.debug(f">>> Received: {msg}")
+        return msg
+
+    async def write(self, data: str) -> None:
+        logger.debug(f"<<< Sending: {data}")
+        await self.write_stream.write(data)
+
+
+def detect_client_type(client_info: dict) -> str:
+    """Detect the type of client connecting to the server."""
+    client_id = client_info.get("client_id", "")
+    client_version = client_info.get("client_version", "")
+
+    # Log more details about the client for troubleshooting
+    logger.debug(
+        f"Client connection details - ID: '{client_id}', Version: '{client_version}'"
+    )
+
+    # More robust detection for Cline VS Code
+    if "cline" in client_id.lower():
```

Comment:
This function uses multiple hard-coded string literals and patterns to detect Cline clients, which is fragile and difficult to maintain. Consider:
1. Creating a `ClientType` enum instead of returning string literals
2. Moving client detection patterns to a configuration file
3. Creating a dedicated module for client detection
4. Using a more extensible pattern matching approach